### PR TITLE
build: pin rmp serde to v0.15.5

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
Pin `rmp-serde` to match the version in `holochain-serialization` and move `rmp-serde`,`lair_keystore`, `lair_keystore_api` and `tx5-signal-srv` to workspace deps.

### TODO:
- [ ] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading

### CODE REVIEW

**Comment by jost-s** [comment]
I want to wait to reference an updated lair once it's available. The current version is still pulling in a different version of rmp-serde.

**Comment by jost-s** [comment]
closes #2990

**Comment by jost-s** [comment]
Now there's only version of `rmp-serde`, `lair_keystore` and `tx5`. Ready for review.

**Review by neonphog — APPROVED** [review]

**Review by ThetaSinner — APPROVED** [review]

File: crates/kitsune_p2p/kitsune_p2p/Cargo.toml [hunk 1]
```diff
@@ -44,7 +44,7 @@ thiserror = "1.0.22"
 tokio = { version = "1.27", features = ["full"] }
 tracing = "0.1"
 tokio-stream = "0.1"
-tx5 = { version = "=0.0.5-alpha", optional = true }
+tx5 = { version = "=0.0.6-alpha", optional = true }
```
Comment (ThetaSinner) [inline crates/kitsune_p2p/kitsune_p2p/Cargo.toml:47]: No workspace version here?

Comment (jost-s) [reply crates/kitsune_p2p/kitsune_p2p/Cargo.toml:47]: It's the only occurrence in the project. Should that become a workspace dep too?

**Review by maackle — APPROVED** [review]

**Review by thedavidmeister — APPROVED** [review]

**Comment by ThetaSinner** [comment]
@jost-s Is this going to be back-ported? I'm wanting Lair on 0.4.1 on `develop-0.2` but I wonder if these changes should come along too in that case?

**Comment by ThetaSinner** [comment]
Cool okay, I'll do the backport because I've already started a PR that isn't building due to dependency conflicts :)

**Pull request merged**